swimlane/core/client.py

```diff
@@ -246,14 +263,50 @@ def user(self):
         """User record instance for authenticated user"""
         return self._session.auth.user
 
+class SwimlaneTokenAuth(SwimlaneResolver):
+    """Handles token authentication for all requests"""
+
+    def __init__(self, swimlane, access_token):
+        super(SwimlaneTokenAuth, self).__init__(swimlane)
+
+        self._access_token = access_token
+        self.user = None
+    
+    def __call__(self, request):
+        """Attach necessary headers to all requests"""
+
+        headers = {
+            'Private-Token': self._access_token
+        }
+
+        request.headers.update(headers)
+
+        # Only make the call to user/authorize to get the user's profile if we haven't retrieved it
+        # already
+        if self.user != None:
+            return request
+
+        # Temporarily remove auth from Swimlane session for auth request to avoid recursive loop during the request
+        self._swimlane._session.auth = None
+        resp = self._swimlane.request(
+            'get',
+            'user/authorize',
+            headers=headers
+        )
+        self._swimlane._session.auth = self
+        
+        json_content = resp.json()
+        self.user = User(self._swimlane, _user_raw_from_login_content(json_content))
+        
+        return request
 
-class SwimlaneAuth(SwimlaneResolver):
+class SwimlaneJwtAuth(SwimlaneResolver):
     """Handles authentication for all requests"""
 
     _token_expiration_buffer = pendulum.Interval(minutes=5)
 
     def __init__(self, swimlane, username, password, verify_ssl=True):
```

Comment:
Looks like I had `verify_ssl` being passed in here previously. I know it used to be needed and can't remember exactly why, but noticed it wasn't in the new `Token` class. If it's not needed anymore, can remove it here, but wanted to double-check it shouldn't also be added in the new class.

Reply:
Good catch, I didn't see any reference to `_verify_ssl` in `client.py` so I went ahead and omitted it in the new class. Looks like `SwimlaneAuth` still works correctly without it so I went ahead and removed it.